### PR TITLE
Added security-related functionality

### PR DESCRIPTION
**Pull request by vikkushnir**
Implemented Role model
created and filled tables like users, books, roles

### CODE REVIEW

File: src/main/java/online/book/store/controller/BookController.java [hunk 1]
```diff
@@ -29,27 +30,31 @@ public class BookController {
     private final BookService bookService;
 
     @GetMapping
+    @PreAuthorize("hasAnyRole('ROLE_ADMIN', 'ROLE_USER')")
```
Comment (ghost) [inline src/main/java/online/book/store/controller/BookController.java]: your user can have many roles. so you don't need to add multiple roles here

File: src/main/java/online/book/store/security/CustomUserDetailsService.java [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+@Service
+@RequiredArgsConstructor
+public class CustomUserDetailsService implements UserDetailsService {
+    private final UserRepository userRepository;
+
+    @Override
+    public UserDetails loadUserByUsername(String email)
+            throws UsernameNotFoundException {
+        return userRepository.findByEmail(email)
+                .orElseThrow(() -> new UsernameNotFoundException(email + " does not exist"));
```
Comment (ghost) [inline src/main/java/online/book/store/security/CustomUserDetailsService.java]: How can you tell if this email exists? =)
Change the message please

File: src/main/resources/db/changelog/changes/04-create-role-table.yaml [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+            columns:
+              - column:
+                  name: id
+                  type: bigint
+                  autoIncrement: true
+                  constraints:
+                    primaryKey: true
+              - column:
+                  name: name
+                  type: enum('ADMIN', 'USER')
```
Comment (ghost) [inline src/main/resources/db/changelog/changes/04-create-role-table.yaml:17]: hardcode is evil)

**Review by ghost — CHANGES_REQUESTED** [review]
GJ!
Some minor comments left)

File: src/main/resources/db/changelog/changes/04-create-role-table.yaml (hunk 3 above)
Comment (vikkushnir) [reply src/main/resources/db/changelog/changes/04-create-role-table.yaml:17]: discussed on QnA, this variant could also be

**Review by andrii-hoienko — APPROVED** [review]
Good job!

**Pull request merged**